_docs/src/documentation/test_projects.md

```diff
@@ -0,0 +1,348 @@
+  }
+
+  protected override void SetupSolutionFolder(RexConfiguration conf, RexTarget target)
+  {
+    base.SetupSolutionFolder(conf, target);
+
+    conf.SolutionFolder = Path.Combine(conf.SolutionFolder, "4_tools");
+  }
+
+  protected override void SetupLibDependencies(RexConfiguration conf, RexTarget target)
+  {
+    base.SetupLibDependencies(conf, target);
+
+    conf.AddPublicDependency<Regina>(target, DependencySetting.Default | DependencySetting.IncludeHeadersForClangtools);
+    conf.AddPublicDependency<RexAutoTest>(target, DependencySetting.Default);
+  }
+
+  protected override void SetupConfigSettings(RexConfiguration conf, RexTarget target)
+  {
+    base.SetupConfigSettings(conf, target);
+
+    conf.Options.Remove(Options.Vc.Compiler.JumboBuild.Enable);
+  }
+}
+```
+
+Next you need to create a source file which will act as the entry point of the auto tests, let's call this file `regina_auto_tests.cpp`
+Fill the file with the following code
+
+```cpp
+// Basic functionality to create auto tests
+#include "rex_auto_test/auto_test.h"
+
+// Entry point we need to define for the rex engine
+#include "rex_engine/entrypoint.h"
+
+// Holds the definition of ApplicationCreationParams
+#include "rex_engine/engine_params.h"
+
+// Includes to query and diagnose the auto test to run
+#include "rex_engine/cmdline.h"
+#include "rex_engine/diagnostics/assert.h"
+
+// Holds the definition of PlatformCreationParams
+#include "rex_windows/platform_creation_params.h"
+
+namespace regina
+{
+  rsl::array g_auto_tests =
+  {
+    // Declare auto tests here
+    // The following is just an example
+    rex::auto_test::AutoTest("Boot", regina_auto_test::boot_test_entry)
+  };
+
+} // namespace regina
+
+namespace rex
+{
+  rex::ApplicationCreationParams app_entry(rex::PlatformCreationParams&& platformParams)
+  {
+    // Get the auto test cmd line argument. This holds the name of the auto test we want to run
+    rsl::optional<rsl::string_view> cmdline = rex::cmdline::get_argument("AutoTest");
+
+    // Make sure it's a known auto test
+    REX_ASSERT_X(cmdline.has_value(), "Auto test fired but no auto test specified on the commandline. Commandline: {}", rex::cmdline::get());
+
+    // Loop through our known auto tests and launch the one we want
+    for (rex::auto_test::AutoTest& auto_test : regina::g_auto_tests)
+    {
+      if (auto_test.is_enabled(cmdline.value()))
+      {
+        return auto_test.launch(rsl::move(platformParams));
+      }
+    }
+
+    // Assert if no valid auto test is found
+    REX_ASSERT("No auto test found for {}", cmdline.value());
+
+    // Return invalid app params if no valid auto test is found
+    return rex::auto_test::invalid_app_params(rsl::move(platformParams));
+  }
+} // namespace rex
+
+```
+
+This will add the boilerplate code that's needed to start the auto tests.
+We added a boot auto test example here, let's dig in to how you'd implement that.
+
+Create a file called `regina_boot_test.cpp`
```

Comment:
A it seems like this is already the case ( this separate file ) why does the user need to create the previous file manually?